File: 프로그래머스/lv2/42885. 구명보트/구명보트.cpp
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<int> people, int limit) {
    int cnt = 0;
    int start = 0, end = people.size() - 1;
    sort(people.begin(), people.end());
    while (start <= end) {
        if (start == end) {
            cnt++;
            break;
        }
        if (people[start] + people[end] <= limit) {
            start++;
        }
        end--;
        cnt++;
    }
    return cnt;
}

File: 프로그래머스/2/42885. 구명보트/구명보트.java
import java.util.*;

class Solution {
    public int solution(int[] people, int limit) {
        Arrays.sort(people);
        int left = 0, right = people.length-1, move = 0;
        while (left <= right) {
            int boat = people[right--];
            if (limit - boat >= people[left]) left++;
            move++;
        }
        return move;
    }
}